README.md
# NAME

Devel::CheckBin - check that a command is available

# SYNOPSIS

    use Devel::CheckBin;

# DESCRIPTION

Devel::CheckBin is a perl module that checks whether a particular command is available.

# USING IT IN Makefile.PL or Build.PL

If you want to use this from Makefile.PL or Build.PL, do not simply copy the module into your distribution as this may cause problems when PAUSE and search.cpan.org index the distro. Instead, use the 'configure\_requires'.



# LICENSE

Copyright (C) tokuhirom

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

# AUTHOR

tokuhirom <tokuhirom@gmail.com>


Changes
Revision history for Perl extension Devel-CheckBin

{{$NEXT}}

0.04 2015-07-22T07:35:08Z

    - Test case fix.
      * win32 fix.
      * added quotes for perl installed into a path with spaces in it
      (mohawk2++)

0.03 2015-04-30T00:47:41Z

    - re-packaging with EU::MM

0.02 2013-08-14T03:17:03Z

    - re-packaging

0.01 2013-04-13T00:20:21Z

    - original version



Build.PL
# =========================================================================
# THIS FILE IS AUTOMATICALLY GENERATED BY MINILLA.
# DO NOT EDIT DIRECTLY.
# =========================================================================

use 5.008_001;

use strict;
use warnings;
use utf8;

use Module::Build;
use File::Basename;
use File::Spec;
use CPAN::Meta;
use CPAN::Meta::Prereqs;

my %args = (
    license              => 'perl',
    dynamic_config       => 0,

    configure_requires => {
        'Module::Build' => 0.38,
    },

    name            => 'Devel-CheckBin',
    module_name     => 'Devel::CheckBin',
    allow_pureperl => 0,

    script_files => [glob('script/*'), glob('bin/*')],
    c_source     => [qw()],

    test_files           => ((-d '.git' || $ENV{RELEASE_TESTING}) && -d 'xt') ? 't/ xt/' : 't/',
    recursive_test_files => 1,
);
if (-d 'share') {
    $args{share_dir} = 'share';
}

my $builder = Module::Build->subclass(
    class => 'MyBuilder',
    code => q{
        sub ACTION_distmeta {
            die "Do not run distmeta. Install Minilla and `minil install` instead.\n";
        }
        sub ACTION_installdeps {
            die "Do not run installdeps. Run `cpanm --installdeps .` instead.\n";
        }
    }
)->new(%args);
$builder->create_build_script();

my $mbmeta = CPAN::Meta->load_file('MYMETA.json');
my $meta = CPAN::Meta->load_file('META.json');
my $prereqs_hash = CPAN::Meta::Prereqs->new(
    $meta->prereqs
)->with_merged_prereqs(
    CPAN::Meta::Prereqs->new($mbmeta->prereqs)
)->as_string_hash;
my $mymeta = CPAN::Meta->new(
    {
        %{$meta->as_struct},
        prereqs => $prereqs_hash
    }
);
print "Merging cpanfile prereqs to MYMETA.yml\n";
$mymeta->save('MYMETA.yml', { version => 1.4 });
print "Merging cpanfile prereqs to MYMETA.json\n";
$mymeta->save('MYMETA.json', { version => 2 });


lib/Devel/CheckBin.pm
package Devel::CheckBin;
use strict;
use warnings;
use 5.008001;
our $VERSION = "0.03";
use parent qw(Exporter);

our @EXPORT = qw(can_run check_bin);

use ExtUtils::MakeMaker;
use File::Spec;
use Config;

# Check if we can run some command
sub can_run {
    my ($cmd) = @_;

    my $_cmd = $cmd;
    return $_cmd if (-x $_cmd or $_cmd = MM->maybe_command($_cmd));

    for my $dir ((split /$Config::Config{path_sep}/, $ENV{PATH}), '.') {
        next if $dir eq '';
        my $abs = File::Spec->catfile($dir, $cmd);
        return $abs if (-x $abs or $abs = MM->maybe_command($abs));
    }

    return;
}

sub check_bin {
    my ( $bin, $version) = @_;
    if ( $version ) {
        die "check_bin does not support versions yet";
    }

    # Locate the bin
    print "Locating bin:$bin...";
    my $found_bin = can_run( $bin );
    if ( $found_bin ) {
        print " found at $found_bin.\n";
        return 1;
    } else {
        print " missing.\n";
        print "Unresolvable missing external dependency.\n";
        print "Please install '$bin' seperately and try again.\n";
        print STDERR "NA: Unable to build distribution on this platform.\n";
        exit(0);
    }
}

1;
__END__

=for stopwords distro

=head1 NAME

Devel::CheckBin - check that a command is available

=head1 SYNOPSIS

    use Devel::CheckBin;

=head1 DESCRIPTION

Devel::CheckBin is a perl module that checks whether a particular command is available.

=head1 USING IT IN Makefile.PL or Build.PL

If you want to use this from Makefile.PL or Build.PL, do not simply copy the module into your distribution as this may cause problems when PAUSE and search.cpan.org index the distro. Instead, use the 'configure_requires'.


=head1 LICENSE

Copyright (C) tokuhirom

This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.

=head1 AUTHOR

tokuhirom E<lt>tokuhirom@gmail.comE<gt>



Makefile.PL
# =========================================================================
# THIS FILE IS AUTOMATICALLY GENERATED BY MINILLA.
# DO NOT EDIT DIRECTLY.
# =========================================================================

use 5.008_001;
use strict;

use ExtUtils::MakeMaker;


my %WriteMakefileArgs = (
    NAME     => 'Devel::CheckBin',
    DISTNAME => 'Devel-CheckBin',
    VERSION  => '0.03',
    CONFIGURE_REQUIRES => {
  "ExtUtils::MakeMaker" => 0
}
,
    BUILD_REQUIRES     => {}
,
    TEST_REQUIRES      => {
  "File::Temp" => 0,
  "Test::More" => "0.98"
}
,
    PREREQ_PM          => {
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => "6.52",
  "parent" => 0,
  "perl" => "5.008001"
}
,
);

my $full_prereqs = {
  "Exporter" => 0,
  "ExtUtils::MakeMaker" => "6.52",
  "File::Temp" => 0,
  "Test::More" => "0.98",
  "parent" => 0,
  "perl" => "5.008001"
}
;

unless (eval { ExtUtils::MakeMaker->VERSION(6.63_03) }) {
    delete $WriteMakefileArgs{TEST_REQUIRES};
    delete $WriteMakefileArgs{BUILD_REQUIRES};
    $WriteMakefileArgs{PREREQ_PM} = $full_prereqs;
}

unless (eval { ExtUtils::MakeMaker->VERSION(6.52) }) {
    delete $WriteMakefileArgs{CONFIGURE_REQUIRES};
}

unless (eval { ExtUtils::MakeMaker->VERSION(6.57_01) }) {
    use File::Copy;
    copy('META.yml', 'MYMETA.yml')   or warn "META.yml: $!";
    copy('META.json', 'MYMETA.json') or warn "META.json: $!";
}

WriteMakefile(%WriteMakefileArgs);
